xio-core/src/test/java/com/xjeffrose/xio/http/GrpcRequestHandlerTest.java

```diff
@@ -0,0 +1,345 @@
+package com.xjeffrose.xio.http;
+
+import com.google.protobuf.GeneratedMessageV3;
+import com.google.protobuf.InvalidProtocolBufferException;
+import helloworld.*;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.UnpooledByteBufAllocator;
+import io.netty.channel.*;
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class GrpcRequestHandlerTest extends Assert {
+
+  private EmbeddedChannel channel;
+  private GrpcRequestHandler subject;
+  private static final String responsePrefix = "I'm a response: ";
+
+  @Before
+  public void setUp() {
+    subject =
+        new GrpcRequestHandler<>(
+            HelloRequest::parseFrom,
+            (HelloRequest request) ->
+                HelloReply.newBuilder().setMessage(responsePrefix + request.getName()).build());
+
+    channel =
+        new EmbeddedChannel(
+            new SimpleChannelInboundHandler<Request>() {
+              @Override
+              protected void channelRead0(ChannelHandlerContext ctx, Request request) {
+                subject.handle(ctx, request, null);
+              }
+            });
+  }
```

Comment:
I like this setup and how the tests are looking for the responses to come out of the handler. excellent!